ludwig/datasets/mixins/load.py

```diff
@@ -48,7 +48,12 @@ def load_processed_dataset(self, split) -> Union[pd.DataFrame,
                 training_set = data_df[data_df[SPLIT] == 0]
                 val_set = data_df[data_df[SPLIT] == 1]
                 test_set = data_df[data_df[SPLIT] == 2]
-                return training_set, test_set, val_set
+                if val_set.shape[0] == 0:
```

Comment:
I would always return 3 elements for consistency. in case `len(*_set) == 0` I would replace it with `None` that is easy to check for.